Apoorv/Apoorv/ViewModels/MainViewModel.cs

```diff
@@ -1,23 +1,75 @@
-﻿using GalaSoft.MvvmLight;
+﻿using System.IO;
+using GalaSoft.MvvmLight;
 using GalaSoft.MvvmLight.Command;
 using Rhino;
 using Rhino.Geometry;
 using Rhino.Input;
 using Rhino.Input.Custom;
+using Grasshopper.Plugin;
 
 namespace Apoorv.ViewModels
 {
     public class MainViewModel : ViewModelBase
     {
         public RelayCommand SelectPointCommand { get; }
+        public RelayCommand OpenGrasshopperCommand { get; }
+        public RelayCommand OpenFileDialog { get; }
         public RhinoDoc Doc { get; set; }
 
         public MainViewModel()
         {
             Doc = RhinoDoc.ActiveDoc;
             SelectPointCommand = new RelayCommand(OnSelectPoint);
+            OpenGrasshopperCommand = new RelayCommand(OnOpenGrasshoper);
+            OpenFileDialog = new RelayCommand(OnOpenFileDialog);
         }
 
+        /// <summary>
+        /// Sets EPW File Path.
+        /// </summary>
+        private void OnOpenFileDialog()
+        {
+            var dialog = new Microsoft.Win32.OpenFileDialog
+            {
+                DefaultExt = ".epw",
+                Filter =
+                    "EPW Files (*.epw)|*.epw"
+            };
+            var result = dialog.ShowDialog();
+            if (result == true)
+            {
+                EpwFilePath = dialog.FileName;
+            }
+        }
+
+        /// <summary>
+        /// Opens Grasshopper and writes a value to it.
+        /// </summary>
+        private void OnOpenGrasshoper()
+        {
+            // (Konrad) This bit might have to change later since we are not likely to publish
+            // this plug-in with that same folder structure. This is solely caused by the fact
+            // that we are debugging this and *bin/ folders are ignored from GitHub.
+            var newPath = string.Empty;
+            var dir = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
+            if (dir != null)
+            {
+                newPath = Path.GetFullPath(Path.Combine(dir, @"..\..\..\")) + @"_Samples\sampleFile.gh";
+            }
+            if (!File.Exists(newPath)) return;
+
+            var gh = RhinoApp.GetPlugInObject("Grasshopper") as GH_RhinoScriptInterface;
+            gh?.OpenDocument(dir);
+            gh?.HideEditor();
+            gh?.AssignDataToParameter("1f808148-f532-4d5a-8a1e-8b0cec03975f", EpwFilePath);
+            gh?.RunSolver(true);
+            gh?.SaveDocument();
+            gh?.CloseDocument();
```

Comment:
so the "?" here is a piece of functionality in C# that is called "null propagation". What that means is basically if `gh == null` don't execute `RunSolver` because that would cause an exception.